toucan_connectors/trello/trello_connector.py

```diff
@@ -0,0 +1,148 @@
+    labels = 'labels'
+
+
+class TrelloDataSource(ToucanDataSource):
+    board_id: str
+    fields_list: List[Fields] = [Fields.name, Fields.url,
+                                 Fields.lists, Fields.members,  Fields.labels]
+    custom_fields: bool = True
+
+
+class TrelloConnector(ToucanConnector):
+    type = "Trello"
+    data_source_model: TrelloDataSource
+
+    key_id: str = ''
+    token: str = ''
+
+    def get_board(self, path, **customParams):
+        return requests.get(f'https://api.trello.com/1/boards/{path}', params={
+            'key': self.key_id,
+            'token': self.token, **customParams
+        }).json()
+
+    def replace_id_by_value(self, card_with_id, lists=None, labels=None,
+                            members=None, custom_fields=None):
+        """
+        `card_with_id` is a dictionary containing all data of a card,
+        but with unreadlable id instead of value
+        This fonction return `card_with_value` with the same dictionnary
+        as `card_with_id` but with readable value
+
+        `lists`: dictionnary of correspondance between list names and ids
+        `members`: dictionnary of correspondance between members names and ids
+        `custom_fields`: dictionnary of correspondance between custom field and there representation
+        """
+        card_with_value = {}
+
+        # id, name and url fields do not need to translate from id to value
+        card_with_value["id"] = card_with_id["id"]
+
+        if 'name' in card_with_id.keys():
+            card_with_value['name'] = card_with_id['name']
+        if 'url' in card_with_id.keys():
+            card_with_value['url'] = card_with_id['url']
+
+        # lists, members and labels need to translate from a id to a value
+        if lists:
+            card_with_value['lists'] = lists[card_with_id['idList']]
+        if members:
+            card_with_value['members'] = [members[member] for member in card_with_id['idMembers']]
+        if labels:
+            card_with_value['labels'] = [labels[label['id']] for label in card_with_id['labels']]
+
+        # custom fields
+        if custom_fields:
+            for card_custom_field in card_with_id['customFieldItems']:
+                custom_field = custom_fields[card_custom_field['idCustomField']]
+                if custom_field['type'] == 'number':
+                    card_with_value[custom_field['name']] = float(
+                        card_custom_field['value']['number'])
+                elif custom_field['type'] == 'text':
+                    card_with_value[custom_field['name']] = card_custom_field['value']['text']
+                elif custom_field['type'] == 'date':
+                    card_with_value[custom_field['name']] = card_custom_field['value']['date']
+                elif custom_field['type'] == 'checkbox':
+                    if card_custom_field['value']['checked'] == 'true':
+                        card_with_value[custom_field['name']] = True
```

Comment:
why is the "unchecked" case not covered ?

Reply:
Because when the case is unchecked the response does not return the customField into the list : card_with_id['customFieldItems'].

Reply:
This means that if I want to put `False` you must do another loop on `custom_fields_id_mapping`

Reply:
ok